nginx/init.sls

```diff
@@ -19,3 +19,20 @@ nginx:
   file.symlink:
     - target: /etc/nginx/sites-available/default
 
+certbot:
+  pkgrepo.managed:
+    - ppa: certbot/certbot
+  pkg.installed:
+    - pkgs:
+      - certbot
+      - python-certbot-nginx
+
+certbot renew:
+  cron.present:
+    - identifier: build-cert-renew
+    - user: root
+    - minute: 0
```

Comment:
In the interest of being a good internet citizen, let's not use midnight directly but a more "random" time as the base - I'm sure they already get way too many requests right at midnight. Additionally, we can use `sleep` to add some more jitter: https://stackoverflow.com/questions/9049460/cron-jobs-and-random-times-within-given-hours/16289693#16289693 for example.